envoy/http/header_evaluator.h

```diff
@@ -16,13 +17,11 @@ class HeaderEvaluator {
    * from the `bar` field in the stream_info, request headers or response headers.
    *
    * @param headers the target header map to be mutated.
-   * @param request_headers request headers to be used in the header manipulation.
-   * @param response_headers response headers to be used in the header manipulation.
+   * @param format_context context to format the header value.
    * @param stream_info the source of values that can be used in the evaluation.
    */
   virtual void evaluateHeaders(Http::HeaderMap& headers,
-                               const Http::RequestHeaderMap& request_headers,
-                               const Http::ResponseHeaderMap& response_headers,
+                               const Formatter::HttpFormatterContext& format_context,
```

Comment:
The http formatter interface is removed, so I also updated the HeaderEvaluator to use new `HttpFormatterContext`.